lib/count_opened_loans/main.py

```diff
@@ -24,57 +26,39 @@ async def get_count_loan_approved_or_disbursed_events(
         utc_endtime: the UTC time to end the time window
 
     Returns:
-        count of LoanApproved or LoanDisbursed events within the given time window
+        count of institutions with one or more opened loans
     """
+    # STEP 1: fetch list of institutions belonging to the user
     institutions = await api_client.institutions.get(user_uuid=user_uuid)
 
-    # subset to only fetch data for institutions known to contain loan-type accounts for the user
-    institutions_w_loan = [
-        inst for inst in institutions if "loan" in inst.account_types
-    ]
-
-    inst_coroutines = [
-        api_client.alerts.get(
-            user_uuid=user_uuid,
-            institution_id=institution.institution_id,
-            utc_starttime=utc_starttime,
-            utc_endtime=utc_endtime,
-            labels=["LoanApproved", "LoanDisbursed"],
+    # subset to only institutions that contain loan-type accounts
+    institutions_w_loan = []
+    for inst in institutions:
+        if "loan" in inst.account_types:
+            institutions_w_loan.append(inst)
+
+    # STEP 2: fetch alert records for all institutions that contain loan-type accounts
+    alerts_coroutines = []
+    for inst_w_loan in institutions_w_loan:
+        alerts_coroutines.append(
+            api_client.alerts.get(
+                user_uuid=user_uuid,
+                institution_id=inst_w_loan.institution_id,
+                utc_starttime=utc_starttime,
+                utc_endtime=utc_endtime,
+                labels=["LoanApproved", "LoanDisbursed"],
+            )
         )
-        for institution in institutions_w_loan
-    ]
-
-    r = await asyncio.gather(*inst_coroutines)
-    institution_ids = [institution.institution_id for institution in institutions]
-
-    alerts_by_institution_id = dict(zip(institution_ids, r))
-
-    alerts_list_by_institution_id: Dict[str, List[Dict]] = {}
-    for institution_id, alerts in alerts_by_institution_id.items():
-        if alerts:
-            alerts_list_by_institution_id[institution_id] = [
-                alert.dict() for alert in alerts
-            ]
-
-    count_loan_approved_or_disbursed_events = 0
-
-    for institution_id, alerts_list in alerts_list_by_institution_id.items():
-
-        filtered_alerts_list_on_time_window = []
 
-        for alert_record in alerts_list:
-            if (
-                alert_record["ts"] >= utc_starttime.timestamp()
-                and alert_record["ts"] < utc_endtime.timestamp()
-            ):
-                filtered_alerts_list_on_time_window.append(alert_record)
+    r = await asyncio.gather(*alerts_coroutines)
```

Comment:
nit: not a fan of one letter variable names. Maybe `alerts_per_institution`?

Reply:
somewhat agreed. I'm OK with one letter vars for "disposable" variables that don't wander far from where they're declared.... more common in C, Go, ... maybe not Python? Either way, that argument doesn't apply to this var; a named variable would be helpful here as it's an important var.  I'l leave this for someone to pick up in a future PR, however, as it should probably be addressed across all features that use a generic `r` as a var for the async gather responses.